Classes/BEMCheckBox.swift

```diff
@@ -0,0 +1,621 @@
+        return frame.size
+    }
+    
+    // MARK: Setters
+    
+    private var _on: Bool = false
+    
+    internal func setOn(_ on: Bool, animated: Bool, notifyGroup: Bool) {
+        self._on = on
+
+        if animated {
+            setupLayers()
+        }
+        else {
+            setNeedsLayout()
+        }
+
+        if on {
+            if animated {
+                addOnAnimation()
+            }
+        } else {
+            if animated {
+                addOffAnimation()
+            } else {
+                onBoxLayer?.removeFromSuperlayer()
+                onBoxLayer = nil
+                
+                checkMarkLayer?.removeFromSuperlayer()
+                checkMarkLayer = nil
+            }
+        }
+
+        if notifyGroup {
+            group?.notifyCheckBoxSelectionChanged(self)
+        }
+    }
+    
+    /** Set the state of the check box to On or Off, optionally animating the  */
+    @objc public func setOn(_ on: Bool, animated: Bool = false) {
+        setOn(on, animated: animated, notifyGroup: false)
+    }
+    
+    // MARK: Gesture Recognizer
+    @objc func handleTapCheckBox(_ recognizer: UITapGestureRecognizer?) {
+        // If we have a group that requires a selection, and we're already selected, don't allow a deselection
+        if group != nil && group?.mustHaveSelection == true && on {
+            return
+        }
+
+        setOn(!on, animated: true)
+        if delegate?.responds(to: #selector(BEMCheckBoxDelegate.didTap(_:))) == true {
+            delegate?.didTap?(self)
+        }
+        sendActions(for: .valueChanged)
+    }
+
+    // MARK: - Helper methods -
+
+    // MARK: Increase touch area
+    
+    override public func point(inside point: CGPoint, with event: UIEvent?) -> Bool {
+        var found = super.point(inside: point, with: event)
+
+        let minimumSize = minimumTouchSize
+        let width = bounds.width
+        let height = bounds.height
+
+        if found == false && (width < minimumSize.width || height < minimumSize.height) {
+            let increaseWidth = minimumSize.width - width
+            let increaseHeight = minimumSize.height - height
+
+            let rect = bounds.insetBy(dx: -increaseWidth / 2, dy: -increaseHeight / 2)
+
+            found = rect.contains(point)
+        }
+
+        return found
+    }
+
+    // MARK: Drawings
+
+    /** Draws the entire checkbox, depending on the current state of the on property. */
+    private func setupLayers() {
+        if !hideBox {
+            setupBoxOffLayer()
+        } else {
+            offBoxLayer?.removeFromSuperlayer()
+            offBoxLayer = nil
+        }
+        
+        if !hideBox && on {
+            setupBoxOnLayer()
+        }
+        
+        if on {
+            setupCheckmarkLayer()
+        }
+    }
+
+    /** Draws the box used when the checkbox is set to Off. */
+    private func setupBoxOffLayer() {
+        if offBoxLayer == nil
+        {
```

Comment:
For consistency, we should open the curly bracket on the same line as the `if`.

Reply:
Tried to keep your styling of curly braces in first line, but when I got to solving the old issue with the layers and rendering - I fell back to my healthy habits of nice and clean curlies ;-)